engines/stark/ui/world/gamewindow.cpp

```diff
@@ -238,4 +241,25 @@ void GameWindow::reset() {
 	_objectRelativePosition.y = 0;
 }
 
+void GameWindow::onScreenChanged() {
+	// May be called when resources have not been loaded
+	if (!StarkGlobal->getCurrent()) {
+		return;
+	}
+
+	Common::Array<Resources::Layer *> layers = StarkGlobal->getCurrent()->getLocation()->listLayers();
+
+	VisualText *text = nullptr;
+	Gfx::RenderEntryArray renderEntries;
```

Comment:
Please reduce the scope of these variables. That'd makes the code a bit easier to read, IMO.

Reply:
Do you mean putting the declaration into the loop? I am not sure how compilers can optimize this these days but isn't that may potentially cause unnecessary variable creations and deallocations, especially since we have an array here?

Reply:
C++ compilers are very good with these little details. For non performance critical code it's better to focus on writing simple and readable code.

In this particular case, constructing `RenderEntryArray` outside of the loop may prevent Return Value Optimization.